be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/RollCallValidator.scala

```diff
@@ -25,14 +28,21 @@ case object RollCallValidator extends MessageDataContentValidator with EventVali
         val laoId: Hash = rpcMessage.extractLaoId
         val expectedRollCallId: Hash = Hash.fromStrings(EVENT_HASH_PREFIX, laoId.toString, data.creation.toString, data.name)
 
+        val sender: PublicKey = message.sender
+        val channel: Channel = rpcMessage.getParamsChannel
+
         if (!validateTimestampStaleness(data.creation)) {
           Right(validationError(s"stale 'creation' timestamp (${data.creation})"))
         } else if (!validateTimestampOrder(data.creation, data.proposed_start)) {
           Right(validationError(s"'proposed_start' (${data.proposed_start}) timestamp is smaller than 'creation' (${data.creation})"))
         } else if (!validateTimestampOrder(data.proposed_start, data.proposed_end)) {
           Right(validationError(s"'proposed_end' (${data.proposed_end}) timestamp is smaller than 'proposed_start' (${data.proposed_start})"))
         } else if (expectedRollCallId != data.id) {
-          Right(validationError("unexpected id"))
+          Right(validationError(s"unexpected id"))
+        } else if (!validateOwner(sender, channel)){
+          Right(validationError(s"invalid sender $sender"))
+        } else if (!validateChannelType(ObjectType.LAO, channel)) {
+          Right(validationError(s"trying to send an OpenRollCall message on a wrong type of channel $channel"))
```

Comment:
Did you mean "trying to send an CreateRollCall message on a wrong type of channel $channel"
Note the cascading IF-ELSE again